include/kernel.h

```diff
@@ -331,6 +333,16 @@ struct _thread_stack_info {
 typedef struct _thread_stack_info _thread_stack_info_t;
 #endif /* CONFIG_THREAD_STACK_INFO */
 
+#if defined(CONFIG_MEMORY_DOMAIN)
```

Comment:
Really do not want CONFIG_MEMORY_DOMAIN to exist.
Have this all depend on CONFIG_USERSPACE.
I know on ARM you don't have userspace implemented yet, just stub out the required functions for now.